forc-plugins/forc-client/src/op/submit.rs

```diff
@@ -1,12 +1,22 @@
 use crate::{cmd, util::node_url::get_node_url};
 use anyhow::Context;
+use forc_tracing::println_warning;
+use forc_util::if_cli_test;
 use fuel_core_client::client::{types::TransactionStatus, FuelClient};
 use fuel_crypto::fuel_types::canonical::Deserialize;
 
 /// A command for submitting transactions to a Fuel network.
 pub async fn submit(cmd: cmd::Submit) -> anyhow::Result<()> {
     let tx = read_tx(&cmd.tx_path)?;
     let node_url = get_node_url(&cmd.network.node, &None)?;
+    if_cli_test! {
```

Comment:
I think you can do this without an env variable ([link](https://stackoverflow.com/questions/24396293/is-it-possible-to-conditionally-compile-a-code-block-inside-a-function))

```suggestion
    #[cfg(test)]
    {
```

Reply:
Even if you pass `cargo run -f test`? That should make it aware of the feature.